components/ai_chat/browser/ai_chat_tab_helper.h

```diff
@@ -102,13 +102,12 @@ class AIChatTabHelper : public content::WebContentsObserver,
   void OnTabContentRetrieved(int64_t for_navigation_id,
                              std::string contents_text,
                              bool is_video = false);
-  void OnAPIStreamDataReceived(int64_t for_navigation_id,
-                               data_decoder::DataDecoder::ValueOrError result);
-  void OnAPIStreamDataComplete(int64_t for_navigation_id,
-                               api_request_helper::APIRequestResult result);
-  void OnAPISuggestedQuestionsResponse(
-      int64_t for_navigation_id,
-      api_request_helper::APIRequestResult result);
+  void OnEngineCompletionDataReceived(int64_t for_navigation_id,
+                                      std::string result);
+  void OnEngineCompletionComplete(int64_t for_navigation_id,
```

Comment:
`OnEngineGenerationComplete` - is it generating engines? :smile:

In thinking about it, we want this to be clear it's the callbacks for `EngineConsumer::SubmitHumanInput` specifically. So I guess my question for you is your opinion on these function names in `EngineConsumer`:
- GenerateQuestionSuggestions
- SubmitHumanInput

Perhaps the latter should be `GenerateAssistantResponse`? In which case we can name these `OnAssistantResponseData` and `OnAssistantResponseComplete`?

Reply:
The active verb that sounds a bit off, as the client is merely getting the response rather than generating it. But don't have a strong opinion about this. `GenerateAssistantResponse` would follow the suggested question function name and agree it would sound clearer. 

Another option could be using `Chat()` (and `OnChatDataReceived/Complete`) and `SuggestedQuestion()` (nit from QuestionSuggestions 😅). PS: was reading `OnEngineGenerationComplete` as "engine's generation"

Reply:
The consumer doesn't know that the class is working over remote API, it could be local for all it cared. But something is generating the questions and the responses, and the consumer is asking for that to happen, so I think it makes sense to go with that new version.

Reply:
Actually I went for a hybrid approach, since these callbacks _are_ actually used by all API calls (inside `RemoteCompletionClient`):

`GenerationResult`
`GenerationDataCallback`
`GenerationCompletedCallback`

It also was a more concise option.